manifests/plugins.pp

```diff
@@ -0,0 +1,92 @@
+# @summary Manage Sensu plugins
+#
+# Class to manage the Sensu plugins.
+#
+# @example
+#   class { 'sensu::plugins':
+#     plugins    => ['disk-checks'],
+#     extensions => ['graphite'],
+#   }
+#
+# @example
+#   class { 'sensu::plugins':
+#     plugins    => {
+#       'disk-checks' => { 'version' => 'latest' },
+#     },
+#     extensions => {
+#       'graphite' => { 'version' => 'latest' },
+#     },
+#   }
+#
+# @param package_ensure
+#   Ensure property for sensu plugins package.
+# @param package_name
+#   Name of the Sensu plugins ruby package.
+# @param dependencies
+#   Package dependencies needed to install plugins and extensions.
+#   Default is OS dependent.
+# @param plugins
+#   Plugins to install
+# @param extensions
+#   Extensions to install
+#
+class sensu::plugins (
+  String $package_ensure = 'installed',
+  String $package_name = 'sensu-plugins-ruby',
+  Array $dependencies = [],
```

Comment:
We should not use empty assignments such as `''`, `[]` and `{}` and instead use `undef`.

Reply:
So using `undef` would require nesting a lot of logic inside an `if` statements which could make the rest of manifest a bit harder to read without changing behavior.  Still want `undef` as default?  Also in case of `$dependencies` the real default is specified in Hiera data.

Reply:
Yes, please use `undef` and update the data types. I think it will only introduce one layer of an `if` statement around the current code.

Reply:
Updated to default to `undef`.